### PR TITLE
Phase 52.5.7: Move ingestion and evidence modules

### LINKED ISSUE #1091: Phase 52.5.7: Move ingestion and external-evidence modules into domain packages
## Summary

Move ingestion, case workflow, evidence linkage, and external-evidence modules into domain packages while preserving evidence authority boundaries.

Part of: #1084
Depends on: #1090
Parallelizable: No

## Execution order

7 of 10

## Scope

- Move detection lifecycle, detection helpers, native detection context, case workflow, and evidence linkage modules into `ingestion` where classified.
- Move external evidence boundary, endpoint, facade, MISP, osquery, and endpoint evidence helpers into `evidence` where classified.
- Preserve legacy import paths through shims.
- Keep adapter modules stable unless the layout contract explicitly classifies a helper move.
- Extend compatibility verifier for old and new ingestion/evidence paths.

## Scope boundary

Do not change alert admission, case linkage, evidence linkage, external evidence enrichment, optional evidence handling, or adapter behavior. Do not promote optional evidence into authority.

## Authority boundary

External evidence, endpoint context, MISP, osquery, and adapter outputs remain subordinate evidence. AegisOps records remain authoritative for admission, case, recommendation, approval, action, reconciliation, and closeout truth.

## Acceptance criteria

- [ ] Ingestion modules live under `ingestion` with legacy import shims.
- [ ] External-evidence modules live under `evidence` with legacy import shims.
- [ ] Ingestion and external evidence tests pass.
- [ ] Import compatibility verifier covers old and new ingestion/evidence paths.
- [ ] Authority-boundary negative tests still pass.

## Verification

- Run phase 25/28 external evidence and osquery/MISP focused tests.
- Run service persistence ingest/case lifecycle tests.
- Run cross-boundary negative e2e tests.
- Run import compatibility verifier.
- Run issue-lint for this issue.

## Negative tests

- Optional external evidence becoming authoritative fails verification.
- Missing legacy ingestion/evidence imports fail verification.
- Case/evidence linkage behavior changes without focused tests fail verification.

## Closeout evidence

Closeout must include moved module list, shim list, ingestion/evidence focused test output, and authority-boundary notes.

### PR DESCRIPTION
**Pull request by TommyKammy**
## Summary
- Move ingestion lifecycle, case workflow, and evidence linkage modules into `aegisops_control_plane.ingestion` with legacy root shims.
- Move external-evidence boundary/facade/MISP/osquery/endpoint helpers into `aegisops_control_plane.evidence` with legacy root shims.
- Extend import compatibility, package scaffold, and layout inventory verification for old/new paths.

## Verification
- `python3 -m unittest control-plane.tests.test_service_persistence_ingest_case_lifecycle control-plane.tests.test_phase28_external_evidence_boundary_refactor control-plane.tests.test_phase28_endpoint_evidence_pack_validation control-plane.tests.test_phase28_misp_enrichment_validation control-plane.tests.test_phase25_osquery_host_context_validation control-plane.tests.test_cross_boundary_negative_e2e_validation`
- `bash scripts/verify-phase-52-5-1-control-plane-layout-inventory-contract.sh`
- `bash scripts/test-verify-phase-52-5-1-control-plane-layout-inventory-contract.sh`
- `bash scripts/verify-phase-52-5-2-package-scaffolding.sh`
- `bash scripts/test-verify-phase-52-5-2-package-scaffolding-and-shim-policy.sh`
- `bash scripts/verify-phase-52-5-2-import-compatibility.sh`
- `bash scripts/verify-phase-52-5-2-layout-guardrail.sh`
- `git diff --check`
- `node <codex-supervisor-root>/dist/index.js issue-lint 1091 --config <supervisor-config-path>`

## Authority boundary
External evidence, endpoint context, MISP, osquery, and adapter outputs remain subordinate evidence. The move is package ownership and import compatibility only; it does not change admission, case, recommendation, approval, action, reconciliation, or closeout authority.

Part of #1084.
Closes #1091.

## Summary by CodeRabbit

* **New Features**
  * External evidence attachments (MISP, osquery, endpoint) with endpoint collection and artifact ingest/backfill.
  * Expanded case workflow APIs for observations, leads, recommendations, handoffs, and dispositions.
  * Enhanced detection ingestion with native detection context attachment and alert→case promotion.

* **Refactor**
  * Reorganized detection, case workflow, and evidence code into dedicated ingestion/evidence packages with legacy import compatibility.

* **Tests**
  * New end-to-end and validation tests for ingest auth, MISP/osquery/endpoint flows, artifact ordering, and rejection cases.

* **Documentation**
  * ADR updates clarifying package layout and compatibility-shim policy.

### CODE REVIEW

**Comment by TommyKammy** [comment]
Tracked PR head `9458f519185e17ba629606e3c0272a59022f1f89` is still draft because provider review is intentionally suppressed.

- reason code: `draft_review_provider_suppressed`
- what is happening: configured provider review stays suppressed until this PR is ready for review.
- automatic retry: yes
- next action: keep the tracked workspace moving toward ready-for-review promotion for PR #1101; the supervisor will retry automatically on later cycles.

GitHub checks may still be pending because external review-provider work does not start while the PR remains draft.

File: control-plane/aegisops_control_plane/evidence/external_evidence_facade.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            result_kind=result_kind,
+            rows=rows,
+            collected_at=collected_at,
+            reviewed_by=reviewed_by,
+            source_id=source_id,
+            collection_path=collection_path,
+            query_context=query_context,
+            evidence_id=evidence_id,
+            observation_scope_statement=observation_scope_statement,
+            observation_id=observation_id,
```
[the remark this answers is not in the document]
Comment (TommyKammy) [reply control-plane/aegisops_control_plane/evidence/external_evidence_facade.py:44]: The supervisor reprocessed this configured-bot finding on the current head `a7d3a56349bc46cd8aa55f6725e9c0db5c5eac89` and classified it as stale.

Evidence: reviewer=coderabbitai file=control-plane/aegisops_control_plane/evidence/external_evidence_facade.py line=44 processed_on_current_head=yes. Source: https://github.com/TommyKammy/AegisOps/pull/1101#discussion_r3173501208

Under the configured `reply_and_resolve` policy, the supervisor is auto-resolving this stale thread now.
